dymos/utils/misc.py

```diff
@@ -74,15 +72,15 @@ def get_state_targets(ode, state_name, state_options):
     """
     ode_inputs = [opts['prom_name'] for (k, opts) in ode.get_io_metadata(iotypes=('input',)).items()]
 
-    if state_options['targets'] is _unspecified:
-        if state_name in ode_inputs:
-            targets = [state_name]
+    if user_targets is _unspecified:
+        if name in ode_inputs:
+            targets = [name]
         else:
             targets = []
-    elif state_options['targets'] is None:
+    elif user_targets is None:
         targets = []
-    elif state_options['targets']:
-        targets = state_options['targets']
+    elif user_targets:
+        targets = user_targets
         if isinstance(targets, str):
             targets = [targets]
     else:
```

Comment:
I think you can get rid of `elif user_targets is None`, and let it fall in to the `else` branch, since they do the same.

Reply:
I think I've simplified this as much as possible without breaking the behavior.